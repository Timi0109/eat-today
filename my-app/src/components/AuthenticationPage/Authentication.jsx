import React, { useState } from "react";
import Title from "../Title.jsx";
import styles from "./Authentication.module.css";
import MenuBar from "../MenuBar/MenuBar.jsx";
import {useHistory} from "react-router-dom";
import { TextField } from '@material-ui/core';
import Footer from "../Footer/Footer.jsx";

const Authentication = () => {
  const [page, setPage] = useState({ isLoginPage: true });
  const [email, setEmail] = useState();
  const [password, setPassWord] = useState();
  const [username, setUserName] = useState();
  
  let history = useHistory();
  const handleLogIn = () => {
    history.push({pathname:`/login/${username}`, state:{name:username}});
  };
  
  const handleSignUp = () => {
    if(!email.includes("@") || !email.includes(".")){
      alert("Email format are invalid!");
      return;
    }
    history.push("/");
  };

  let content;
  if (page.isLoginPage) {
    content = (
      <div className={styles.box}>
        <form onSubmit={() => handleLogIn()}> 
          <input className={styles.username} placeholder="Username" onChange={(e) => setUserName(e.target.value)} required />
          <br></br>
          <input className={styles.password} type="password" placeholder="Password" onChange={(e) => setPassWord(e.target.value)} required />
          <br></br>
          <button className={styles.loginBtn} type="submit"> LOGIN </button>
        </form>
      </div>
    );
  } else {
    content = (
      <div className={styles.box}>
        <form onSubmit={() => handleSignUp()}>
          <input className={styles.username} placeholder="Username" onChange={(e) => {setUserName(e.target.value)}} required />
          <br></br>
          <input className={styles.emailInput} placeholder="test@example.com" onChange={(e) => {setEmail(e.target.value)}} required />
          <br></br>
          <input className={styles.password} type="password" placeholder="Password" onChange={(e) => {setPassWord(e.target.value)}} required/>
          <br></br>
          <button className={styles.signUpBtn} type="submit"> SIGN UP </button>
        </form>
      </div>
    );
  }
  return (
    <div className={styles.main}>
        <div className={styles.menu}>
            <MenuBar ></MenuBar>
            <Title></Title>
        </div>

        <div className={styles.content}>
            <div className={styles.tabContainer}>
                <button
                className={page.isLoginPage ? styles.signUpTab : styles.signUpTabActive}
                onClick={() => setPage({ isLoginPage: false })}
                >
                Sign Up
                </button>
                <button
                className={page.isLoginPage ? styles.logInTabActive : styles.logInTab}
                onClick={() => setPage({ isLoginPage: true })}
                >
                Login
                </button>
            </div>
            {content}
        </div>
        <Footer></Footer>
    </div>
  );
};



export default Authentication;
