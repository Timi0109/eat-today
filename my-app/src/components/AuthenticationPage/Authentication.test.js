import React from "react";
import Authentication from "./Authentication";

describe("Authentication tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Authentication></Authentication>);
    expect(container).toMatchSnapshot();
  });

});
