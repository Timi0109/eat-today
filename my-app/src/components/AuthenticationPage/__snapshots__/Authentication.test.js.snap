// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication tests renders correctly 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="menu"
    >
      <nav
        class="Nav"
      >
        <a
          class="Nav-child"
          href="/"
        >
          Home
        </a>
        <a
          class="Nav-child"
          href="/recipes"
        >
          Recipe
        </a>
        <a
          class="Nav-child"
          href="/login"
        >
          LogIn
        </a>
      </nav>
      <h2
        class="title"
      >
        EatToday
      </h2>
    </div>
    <div
      class="content"
    >
      <div
        class="tabContainer"
      >
        <button
          class="signUpTab"
        >
          Sign Up
        </button>
        <button
          class="logInTabActive"
        >
          Login
        </button>
      </div>
      <div
        class="box"
      >
        <form>
          <input
            class="username"
            placeholder="Username"
            required=""
          />
          <br />
          <input
            class="password"
            placeholder="Password"
            required=""
            type="password"
          />
          <br />
          <button
            class="loginBtn"
            type="submit"
          >
             LOGIN 
          </button>
        </form>
      </div>
    </div>
    <div
      class="footer"
    >
      <p
        class="title"
      >
        EATTODAY.COM
      </p>
      <div
        class="container"
      >
        <div>
          <p
            class="subTitle"
          >
            About Us
          </p>
          <p>
            Home
          </p>
          <p>
            How It Works
          </p>
        </div>
        <div>
          <p
            class="subTitle"
          >
            Contact Us
          </p>
          <p>
            Linkedin
          </p>
          <p>
            Twitter
          </p>
          <p>
            Wechat
          </p>
        </div>
      </div>
      <p
        class="right"
      >
        @2020 EATTODAY.
      </p>
    </div>
  </div>
</div>
`;

exports[`FoodContent tests renders correctly 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="menu"
    >
      <nav
        class="Nav"
      >
        <a
          class="Nav-child"
          href="/"
        >
          Home
        </a>
        <a
          class="Nav-child"
          href="/recipes"
        >
          Recipe
        </a>
        <a
          class="Nav-child"
          href="/login"
        >
          LogIn
        </a>
      </nav>
      <h2
        class="title"
      >
        EatToday
      </h2>
    </div>
    <div
      class="content"
    >
      <div
        class="tabContainer"
      >
        <button
          class="signUpTab"
        >
          Sign Up
        </button>
        <button
          class="logInTabActive"
        >
          Login
        </button>
      </div>
      <div
        class="box"
      >
        <form>
          <input
            class="username"
            placeholder="Username"
            required=""
          />
          <br />
          <input
            class="password"
            placeholder="Password"
            required=""
            type="password"
          />
          <br />
          <button
            class="loginBtn"
            type="submit"
          >
             LOGIN 
          </button>
        </form>
      </div>
    </div>
    <div
      class="footer"
    >
      <p
        class="title"
      >
        EATTODAY.COM
      </p>
      <div
        class="container"
      >
        <div>
          <p
            class="subTitle"
          >
            About Us
          </p>
          <p>
            Home
          </p>
          <p>
            How It Works
          </p>
        </div>
        <div>
          <p
            class="subTitle"
          >
            Contact Us
          </p>
          <p>
            Linkedin
          </p>
          <p>
            Twitter
          </p>
          <p>
            Wechat
          </p>
        </div>
      </div>
      <p
        class="right"
      >
        @2020 EATTODAY.
      </p>
    </div>
  </div>
</div>
`;
